client/transaction-pool/src/lib.rs

```diff
@@ -130,3 +175,158 @@ impl<PoolApi, Block> TransactionPool for BasicPool<PoolApi, Block>
 		self.pool.on_broadcasted(propagations)
 	}
 }
+
+#[cfg_attr(test, derive(Debug))]
+enum RevalidationStatus<N> {
+	/// The revalidation has never been completed.
+	NotScheduled,
+	/// The revalidation is scheduled.
+	Scheduled(Option<std::time::Instant>, Option<N>),
+	/// The revalidation is in progress.
+	InProgress,
+}
+
+enum RevalidationStrategy<N> {
+	Always,
+	Light(RevalidationStatus<N>)
+}
+
+impl<N: Clone + Copy + SimpleArithmetic> RevalidationStrategy<N> {
+	pub fn clear(&mut self) {
+		if let Self::Light(status) = self { status.clear() }
+	}
+
+	pub fn resubmit_required(&mut self) -> bool {
+		if let Self::Light(_) = self { return false } else { return true }
+	}
+
+	pub fn is_required(
+		&mut self,
+		block: N,
+		revalidate_time_period: Option<std::time::Duration>,
+		revalidate_block_period: Option<N>,
+	) -> bool {
+		if let Self::Light(status) = self {
+			status.is_required(block, revalidate_time_period, revalidate_block_period)
+		} else { true }
```

Comment:
If we introduce the background thread, revalidating front of the ready queue will not be required for `Full` mode.